tests/periph_i2c/tests/03__periph_i2c_error_codes.robot

```diff
@@ -0,0 +1,26 @@
+*** Settings ***
+Documentation       Tests to verify correct error codes are given.
+
+Suite Setup         Run Keywords    Reset DUT and PHILIP
+...                                 API Firmware Should Match
+Test Setup          Run Keywords    Reset DUT and PHILIP
+...                                 API Firmware Should Match
+...                                 I2C Acquire
+Test Teardown       I2C Release
+
+Resource            periph_i2c.keywords.txt
+Resource            api_shell.keywords.txt
+
+Force Tags          periph  i2c
+
+*** Test Cases ***
+Invalid Address Should Fail With Address NACK
+    [Documentation]                         Verify address NACK (ENXIO) occurs
+    API Call Should Error                   I2C Read Reg  addr=42
+    API Last Result Message Should Contain  ENXIO
+
+Invalid Data Should Fail With Data NACK
+    [Documentation]                         Verify data NACK (EIO) occurs
+    PHiLIP.write and execute                i2c.mode.nack data  1
```

Comment:
regarding the error: I think there is space missing between nack and data, should be 
```
i2c.mode.nack  data  1
```